src/probnum/diffeq/odefilter/init_routines/_autodiff.py

```diff
@@ -0,0 +1,247 @@
+"""Automatic-differentiation-based initialization routines."""
+
+import numpy as np
+
+from probnum import problems, randprocs, randvars
+
+from ._interface import InitializationRoutine
+
+
+class _AutoDiffBase(InitializationRoutine):
+    def __init__(self):
+        super().__init__(is_exact=True, requires_jax=True)
+        self._jax, self._jnp = self._import_jax()
+
+    def _import_jax(self):
```

Comment:
these kinds of imports are my attempt at handling the optional JAX dependency cleanly. Once the backend-PR is through, the Jax-handling will be much cleaner anyway.

Reply:
Why don't we move the try-import-except block top-level and -- in AutoDiffBase -- only check whether jax could be imported? I think the solution you propose here is problematic.

Reply:
If I remember correctly, there import error would be raised too early, i.e., when setting up the `init_routines` namespace. In this case, none of the routines could be used. 

I can check again later.